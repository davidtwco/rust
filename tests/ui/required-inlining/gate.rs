// gate-test-required_inlining
//@ compile-flags: --crate-type=lib
#![allow(internal_features)]
#![feature(rustc_attrs)]

#[rustc_no_mir_inline]
#[inline(required)]
//~^ ERROR invalid argument
//~^^ HELP valid inline arguments are `always` and `never`
pub fn bare_required() {
}

#[rustc_no_mir_inline]
#[inline(required("justification"))]
//~^ ERROR invalid argument
//~^^ HELP valid inline arguments are `always` and `never`
pub fn justified_required() {
}

#[rustc_no_mir_inline]
#[inline(must)]
//~^ ERROR invalid argument
//~^^ HELP valid inline arguments are `always` and `never`
pub fn bare_must() {
}

#[rustc_no_mir_inline]
#[inline(must("justification"))]
//~^ ERROR invalid argument
//~^^ HELP valid inline arguments are `always` and `never`
pub fn justified_must() {
}
