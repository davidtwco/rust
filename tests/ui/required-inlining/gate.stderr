error[E0535]: invalid argument
  --> $DIR/gate.rs:6:10
   |
LL | #[inline(required)]
   |          ^^^^^^^^
   |
   = help: valid inline arguments are `always` and `never`

error[E0535]: invalid argument
  --> $DIR/gate.rs:13:10
   |
LL | #[inline(required("justification"))]
   |          ^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: valid inline arguments are `always` and `never`

error[E0535]: invalid argument
  --> $DIR/gate.rs:20:10
   |
LL | #[inline(must)]
   |          ^^^^
   |
   = help: valid inline arguments are `always` and `never`

error[E0535]: invalid argument
  --> $DIR/gate.rs:27:10
   |
LL | #[inline(must("justification"))]
   |          ^^^^^^^^^^^^^^^^^^^^^
   |
   = help: valid inline arguments are `always` and `never`

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0535`.
